pkg/internal/controller/controller.go

```diff
@@ -156,7 +164,7 @@ func (c *Controller) Start(ctx context.Context) error {
 		// caches.
 		for _, watch := range c.startWatches {
 			c.Log.Info("Starting EventSource", "source", watch.src)
-			if err := watch.src.Start(ctx, watch.handler, c.Queue, watch.predicates...); err != nil {
+			if err := watch.src.Start(sourceStartCtx, watch.handler, c.Queue, watch.predicates...); err != nil {
```

Comment:
This context should be set up for each `Start` call. Otherwise there could be an erroneous timeout if you have a bunch of watches.
```suggestion
			// use a context with timeout for launching sources and syncing caches.
			watchStartCtx, cancel := context.WithTimeout(ctx, c.CacheSyncTimeout)
			defer cancel()
			if err := watch.src.Start(watchStartCtx, watch.handler, c.Queue, watch.predicates...); err != nil {
```